server/src/main/java/org/elasticsearch/cluster/metadata/MappingMetadata.java

```diff
@@ -42,10 +44,15 @@ public class MappingMetadata implements SimpleDiffable<MappingMetadata> {
 
     private final boolean routingRequired;
 
+    private final Map<String, Set<String>> fieldsForModels;
```

Comment:
I think this part should be in the `IndexMetadata` directly rather than in the `MappingMetadata`. This way we can ensure that it always exists instead of having an empty map when the mappings is instantiated differently.

Reply:
Yes, this will be addressed by https://github.com/elastic/elasticsearch/pull/103319.